lib/openstack_query/managers/server_manager.py

```diff
@@ -26,168 +27,114 @@ class ServerManager(QueryManager):
-                },
-            ),
-            output_details=output_details,
+        preset_details = QueryPresetDetails(
+            preset=QueryPresetsDateTime.from_string(search_mode),
+            prop=ServerProperties.from_string(property_to_search_by),
+            args={
+                "days": days,
+                "hours": hours,
+                "minutes": minutes,
+                "seconds": float(seconds),
+            },
         )
 
-    def search_servers_last_updated_before_relative_to_now(
-        self,
-        output_details: QueryOutputDetails,
-        days: int,
-        hours: int = 0,
-        minutes: int = 0,
-        seconds: int = 0,
-    ) -> QueryReturn:
-        """
-        method that returns a list of all servers which were last updated before a time relative to now.
-        Uses UTC timezone
-        :param output_details: A dataclass containing config info on how results should be returned
-        :param days: number of days since current time
-        :param hours: number of hours since current time
-        :param minutes: number of minutes since current time
-        :param seconds: number of seconds since current time
-        """
         return self._build_and_run_query(
-            preset_details=QueryPresetDetails(
-                preset=QueryPresetsDateTime.OLDER_THAN,
-                prop=ServerProperties.SERVER_LAST_UPDATED_DATE,
-                args={
-                    "days": days,
-                    "hours": hours,
-                    "minutes": minutes,
-                    "seconds": float(seconds),
-                },
+            preset_details=preset_details,
+            output_details=QueryOutputDetails.from_kwargs(
+                prop_cls=ServerProperties, **kwargs
             ),
-            output_details=output_details,
         )
 
-    def search_servers_last_updated_after_relative_to_now(
-        self,
-        output_details: QueryOutputDetails,
-        days: int,
-        hours: int = 0,
-        minutes: int = 0,
-        seconds: int = 0,
+    def search_by_property(
+        self, search_mode: bool, property_to_search_by: str, values: List[str], **kwargs
     ) -> QueryReturn:
         """
-        method that returns a list of all servers which were last updated after a time relative to now.
-        Uses UTC timezone
-        :param output_details: A dataclass containing config info on how results should be returned
-        :param days: number of days since current time
-        :param hours: number of hours since current time
-        :param minutes: number of minutes since current time
-        :param seconds: number of seconds since current time
+        method that builds and runs a query to find Openstack servers with a selected property
+        matching, or not matching given value(s)
+        :param search_mode: A boolean representing what to return from query, if True - use the preset ANY_IN/EQUAL_TO,
```

Comment:
I'm not keen on using a `bool` to represent a non-true/false condition

Can we just use an enum with the values:
`NOT_ANY_IN`
`NOT_EQUAL_TO`
`ANY_IN`
`EQUAL_TO`

The fact the last two are equal in behaviour at this level is an implementation detail

Reply:
I'm not a fan of using `EQUAL_TO` here - adds unnecessary checks to ensure that user only entered one value - which is handled internally if it were the case
I can change it to `ANY_IN`, `NOT_ANY_IN` - but it seems less intuitive to me than false/true

Reply:
I changed this after talking to Jacob